.github/workflows/CI_Pipeline.yml

```diff
@@ -0,0 +1,43 @@
+name: Continuous Integration
+on: 
+  push:
+    paths-ignore:
+      - 'README.md'
+      - 'Vagrantfile'
+      - 'documentation/**'
+
+  pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - 'README.md'
+      - 'Vagrantfile'
+      - 'documentation/**'
+
+
+jobs:    
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: browser-actions/setup-geckodriver@latest
+    - run: geckodriver --version
+    - uses: actions/checkout@v2
+    - run: docker build --target test --tag my-test-image .
+    - run: docker run --env-file .env.test  my-test-image tests
+    - run: docker run -e TRELLO_KEY=${{secrets.TRELLO_KEY}}  -e TRELLO_TOKEN=${{secrets.TRELLO_TOKEN}} -e TRELLO_BOARD_ID=${{secrets.TRELLO_BOARD_ID}} my-test-image tests_e2e
+
+  deploy:
+    name: Deploy build package to Heroku
+    needs: build
+    if: ${{ github.event_name ==  'push' && github.ref == 'refs/heads/exercise-8' }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2 # Adds a step to checkout the repository code
+    - uses: akhileshns/heroku-deploy@v3.12.12 # This is the action 
+      with:
+        heroku_api_key: ${{secrets.HEROKU_API_KEY}}
+        heroku_app_name: "joe-devops-project" #Must be unique in Heroku
+        heroku_email: "josephbwo@hotmail.com"
+        usedocker: true
```

Comment:
The alternative is a series of 3 or 4 "run" steps. The slight downside to using this Action is the inability to specify build options - see my other comment on documenting the issue.